evoting/lib/chains.go

```diff
@@ -1,22 +1,31 @@
 package lib
 
 import (
-	"github.com/dedis/cothority/skipchain"
 	"github.com/dedis/onet"
-)
 
-var client = skipchain.NewClient()
+	"github.com/dedis/cothority/skipchain"
+)
 
-// New creates a new skipchain for a given roster and stores data in the genesis block.
-func New(roster *onet.Roster, data interface{}) (*skipchain.SkipBlock, error) {
-	return client.CreateGenesis(roster, 1, 1, skipchain.VerificationStandard, data, nil)
+// NewSkipchain creates a new skipchain for a given roster and verification function.
+func NewSkipchain(roster *onet.Roster, verifier []skipchain.VerifierID, data interface{}) (
+	*skipchain.SkipBlock, error) {
+	client := skipchain.NewClient()
+	return client.CreateGenesis(roster, 1, 1, verifier, data, nil)
 }
 
-// chain returns a skipchain for a given id.
-func chain(roster *onet.Roster, id skipchain.SkipBlockID) ([]*skipchain.SkipBlock, error) {
-	chain, err := client.GetUpdateChain(roster, id)
+// Store appends a new block holding data to an existing skipchain.
+func Store(id skipchain.SkipBlockID, roster *onet.Roster, data ...interface{}) error {
+	client := skipchain.NewClient()
+	reply, err := client.GetUpdateChain(roster, id)
 	if err != nil {
-		return nil, err
+		return err
+	}
+
+	for _, d := range data {
+		_, err = client.StoreSkipBlock(reply.Update[len(reply.Update)-1], nil, d)
```

Comment:
According to https://github.com/dedis/cothority/blob/master/skipchain/api.go#L85, this should now be the skipchain-id.

So you won't even need the call to `GetUpdateChain` anymore.

Reply:
I'm not sure how I can store a new block by just using the skipchain ID?
https://github.com/dedis/cothority/blob/master/skipchain/api.go#L85 still takes the latest block
as an argument.